File: VB/Services/ReportProvider.vb
Imports DevExpress.XtraReports.Services
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraReports.Web.WebDocumentViewer
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web

Namespace T132094.Services
    Public Class CustomReportProvider
        Implements DevExpress.XtraReports.Services.IReportProvider

        Public Function GetReport(ByVal id As String, ByVal context As ReportProviderContext) As XtraReport Implements IReportProvider.GetReport
            Select Case id
                Case "DefaultReport"
                    Return New DefaultReport()
                Case "Categories"
                    Return New CategoriesReport()
                Case "Products"
                    Return New ProductsReport()
                Case "Orders"
                    Return New OrdersReport()
                Case Else
                    Return Nothing
            End Select
        End Function
    End Class
End Namespace

File: CS/Services/ReportProvider.cs
using DevExpress.XtraReports.Services;
using DevExpress.XtraReports.UI;
using DevExpress.XtraReports.Web.WebDocumentViewer;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace T132094.Services {
    public class CustomReportProvider : DevExpress.XtraReports.Services.IReportProvider {
        public XtraReport GetReport(string id, ReportProviderContext context) {
            switch (id) {
                case "DefaultReport":
                    return new DefaultReport();
                case "Categories":
                    return new CategoriesReport();
                case "Products":
                    return new ProductsReport();
                case "Orders":
                    return new OrdersReport();
                default:
                    return null;
            }
        }
    }
}
